src/components/common/TheContainer.vue

```diff
@@ -6,7 +6,9 @@
 
 <style lang="scss" scoped>
 .container {
+  max-width: 1440px;
```

Comment:
I don't know exactly what you are planning to do with this component, whether you plan to wrap every App section in it or not, but if this `max-width` will repeat in other components it is a good practice to move this value to SCSS variable, so you don't repeat yourself. 😉

Reply:
Initially I had an intention to use it in the pages components but then moved it to the App.vue file as a general wrapper of all the content. Maybe it's redundant then and instead I could just set max-width of the <main class="app">? What would you suggest?

Reply:
This is generally a cool idea. Unfortunately, it will cause problems in many cases though, depending on the design. Let's imagine most of the layout is inside this `1440px` box, but then you have some full-width banner (100vw) somewhere and now we are in a pretty bad spot, because the whole app works based on global max-width, and you can't just put the banner outside of it without hacks like negative margins. Hacky solutions are not ideal ☹️. Personally, I tend to use a max-width per page basis. This allows to have a lot less code but leaves some space for customization in case some page has wide banners/carousels/etc. in the design. In my recent project, I created a utility class `.boxed-layout` : 
```
.boxed-layout {
  max-width: $layout-width;
  margin: 0 auto;
  padding: 0 24px;
}
```
This works fine and I'm pretty happy with this solution 😉

Reply:
You've catched me -- I was lazy and accepted this little inconsistency with the design for mobile view, where the hero section of the home page should be of width: 100vw. I like a lot your solution with a utility class and will keep it in mind for the future. However, for this project it seems useful to me to keep TheContainer, but to move it to the page level, as you suggested. Moreover, I have introduced a prop `variant` to handle the edge case of hero section.